mne/decoding/time_gen.py

```diff
@@ -625,39 +707,34 @@ def _sliding_window(times, window_params, sfreq):
         List of training slices, indicating for each classifier the time
         sample (in indices of times) to be fitted on.
     """
+    import copy
 
-    window_params = _DecodingTime(window_params)
+    window = _DecodingTime(copy.deepcopy(window))
 
     # Default values
-    if ('slices' in window_params and
-            all(k in window_params for k in
-                ('start', 'stop', 'step', 'length'))):
-        time_pick = window_params['slices']
+    has_slice = 'slices' in window
+    if has_slice:
+        time_pick = window['slices']
```

Comment:
There are two possibilities. Either the user defines the time slices manually, or uses the `start`, `stop`, `step`, `length` parameters to define them. I've tried to clarify the code and internal docs.